app/services/record_participant_declaration.rb

```diff
@@ -10,7 +13,8 @@ def call(params)
   end
 
   def call
-    RecordDeclarations::RecorderFactory.call(course_identifier).call(params)
+    recorder = "::RecordDeclarations::#{::RecordDeclarations::EventFactory.call(event)}::#{::RecordDeclarations::RecorderFactory.call(course_identifier)}".constantize
```

Comment:
I think this should be thoroughly unit tested.

I would also be very tempted to replace `::RecordDeclarations::EventFactory.call(event)` and `::RecordDeclarations::RecorderFactory.call(course_identifier)` with helper methods in this file - I am not sure what value havign them separated in their own files brings. 

It certainly makes it harder to double check that the combinations of values they spit out match the names of the recorders we define.